nodejs/week2/MealSharing/src/backend/routes/meals.js

```diff
@@ -0,0 +1,49 @@
+const meals = require('../data/meals.json');
+
+const express = require('express');
+const router = express.Router();
+
+router.get('/', (request, response) => {
+  response.send(meals);
+});
+router.get('/:id', async (req, res) => {
+  let result = meals.filter((data) => {
+    if (data.id === parseInt(req.params.id)) return true;
+  });
+  if (result.length != 0) {
+    res.send(result);
+  } else {
+    res
+      .status(404) // HTTP status 404: NotFound
+      .send('Not found');
+  }
+});
+
+router.get('/', async (req, res) => {
+  /**maxPrice	Get meals that has a price smaller than maxPrice	Number	/meals?maxPrice=90 */
+  if (req.query.maxPrice) {
+    res.send(meals.filter((data) => data.price < Number(req.query.maxPrice)));
+  }
+
+  /**title	Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde	String	/meals?title=Indian%20platter */
+  if (req.query.title) {
+    res.send(
+      meals.filter((data) => data.title.includes(String(req.query.title)))
+    );
+  }
+
+  /**createdAfter	Get meals that has been created after the date	Date	/meals?createdAfter=2019-04-05 */
+  if (req.query.createdAfter) {
+    res.send(meals.filter((data) => data.createdDate > req.query.createdAfter));
```

Comment:
You can use Date.parse(data.createdDate) and Date.parse(req.query.createdAfter) which will turn the data.createdDate and req.query.createdAfter string to number, so then you will be able to compare them.